### ISSUE TITLE
Is it possible to search when an item is already selected?

### ISSUE THREAD

**Issue by thearchitect3344** [mskocik/svelecte#201]
I wasn't able to get the search function to work when I have already selected an item on svelecte that does not have multiple enabled. It appears that I have to deselect the item via backspace in order to search for a new item. Is it possible to allow searching items while an item is already selected? It would be nice to be able to search for the new item without having to clear out the select every time. A use case for this could be a select that already has an item pre-selected, but we want to allow the user to quickly, easily change it without having to clear it out first.

**Comment by mskocik** [comment]
This behaviour is intentional, because otherwise user can think it is a multiselect. I am not sure I want to support this use case. Because this use case - selecting different item when other is selected is possible, only without keyboard search.

**Comment by thearchitect3344** [comment]
> This behaviour is intentional, because otherwise user can think it is a multiselect. I am not sure I want to support this use case. Because this use case - selecting different item when other is selected is possible, only without keyboard search.

Thanks for your reply. I see your point about thinking it could be a multiselect, but, I think that is alleviated since they would have had to have a single selection done first. It would be nice to have some way to select via keyboard search. The case I needed it in had a lot of options and scrolling to select became kind of cumbersome.

**Comment by mskocik** [comment]
You can always remove current selection by pressing <kbd>Backspace</kbd> or <kbd>Delete</kbd> on your keyboard before starting the search

**Comment by Jojoshua** [comment]
@mskocik IMO this is a poor workaround. If the component claims to be searchable but the user has to hit backspace first that is a bad experience. Could this behavior be allowed if it was known to be a single select?

**Comment by mskocik** [comment]
> @mskocik IMO this is a poor workaround. If the component claims to be searchable but the user has to hit backspace first that is a bad experience. Could this behavior be allowed if it was known to be a single select?

Maybe it can (not sooner than v4). Not sure about required code changes and especially edge cases if there will be any.
